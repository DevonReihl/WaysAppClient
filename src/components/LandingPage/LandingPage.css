.wrapper div {
  padding: 10px;
  max-width: 200px;
}

.landing-page {
  text-align: center;
  padding: 0.5em;
  position: absolute;
  z-index: 1;
  margin-top: 11em;
  margin-right: 0;
  width: 100% !important;
  display: flex;
  flex-direction: column;
  align-items: center;
}

.landing-page-container {
  display: flex;
  justify-content: center;
}

.dash-btn {
  background-color: white !important;
}

.dash-btn:hover {
  color: #e61f82 !important;
}

.ways-header {
  position: absolute;
  z-index: -1;
  font-size: 10vw;
  color: #e61f82;
  font-family: 'Archivo Black', Arial, Helvetica, sans-serif;
  text-shadow: 2px 1px 2px rgb(0, 0, 0);
  letter-spacing: -10px;
  height: auto;
  width: auto;
  margin-top: 2%;
  margin-right: 10%;
}

section.landing-page::before {
  content: '';
  margin-top: 20%;
}

.background {
  position: relative;
  width: 100vw;
  z-index: 1000;
  display: flex;
  justify-content: center;
  height: auto;
  margin-right: 20%;
  display: flex;
  flex-direction: row;
}

.background img {
  margin-bottom: 50px;
  margin-top: 3vh;
  margin-right: 90vw;
  padding-right: 5vw;
  width: 75% !important;
  height: auto;
  position: absolute;
}

.about-ways-container {
  margin: 1em 0 0 0;
  padding: 10px;
  width: 100vw;
  position: relative;
  background-color: #65b2ef;
}

.fa-hotel {
  color: rgb(46, 140, 216);
}

.fa-plane-departure {
  color: #a353d8;
}

.fa-car {
  color: #d33e88 !important;
}

.about-ways {
  display: flex;
  flex-direction: column;
  text-align: left;
  justify-content: center;
  align-items: center;
  padding: 0 15% 0 15%;
}

.wrapper {
  position: relative;
  background-color: white;
  color: rgb(53, 53, 53);
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  width: 100vw;
  font-size: 20px;
  font-weight: 900;
  padding-top: 20px;
}

.lets-go-btn {
  box-shadow: none;
  margin-top: 20px;
  text-align: center;
  width: 180px;
  background-color: white;
}

.lets-go-btn:hover {
  color: #e61f82;
}

.wrapper hr {
  border: 1px solid rgb(46, 140, 216);
}

.background-image {
  position: absolute;
  opacity: 0.8;
  width: 100vw;
  height: 100vh;
}

.landing-page-background {
  position: absolute;
  width: 100vw;
  height: 100%;
  margin: 0;
  padding: 0;
}

body {
  width: 100vw;
  height: auto;
  margin: 0;
  padding: 0;
  overflow-x: hidden;
}

.landing-page-container-1 {
  padding: 0;
  min-height: 1000px;
  width: 100vw;
}

.longFade {
  animation: fadeIn;
  animation-duration: 1.5s;
}

@media (max-height: 1000px) {
  .background-image {
    min-height: 1000px !important;
  }
}

@media (max-width: 349px) {
  .landing-page-container-1 {
    height: 970px !important;
  }
  .ways-header {
    font-size: 19vw !important;
    letter-spacing: -3px !important;
  }

  .wrapper {
    background-color: white;
    display: flex;
    flex-direction: column !important;
    justify-content: space-around;
    align-items: center;
    width: 100vw;
    font-size: 20px;
    font-weight: 500;
    padding-top: 20px;
  }

  .linkToDashButton {
    display: flex !important;
    flex-direction: column !important;
    align-content: center !important;
    justify-content: center !important;
    background-color: #cce8ff;
    width: 100%;
    margin-top: 0 !important;
    margin-bottom: 0 !important;
    padding-bottom: 50px;
    padding-top: 15px;
    vertical-align: middle;
    height: 100%;
  }

  .landing-page-background {
    margin: 0 !important;
  }
}

@media (max-width: 530px) {
  .wrapper {
    display: flex;
    flex-direction: column !important;
    align-items: center;
    width: 100vw !important;
  }

  .linkToDashButton {
    width: 100vw !important;
  }
}

@media (max-width: 449px) and (min-width: 350px) {
  .ways-header {
    font-size: 16vw !important;
    letter-spacing: -2px !important;
  }
}

@media (max-width: 600px) and (min-width: 450px) {
  .ways-header {
    font-size: 14vw !important;
    letter-spacing: -3px !important;
  }

  .background img {
    width: 40% !important;
  }
}

@media (max-width: 600px) {
  .background img {
    width: 70% !important;
  }
}

@media (max-width: 920px) and (min-width: 600px) {
  .ways-header {
    font-size: 12vw !important;
    letter-spacing: -5px !important;
  }
  .background img {
    width: 70% !important;
  }
}
