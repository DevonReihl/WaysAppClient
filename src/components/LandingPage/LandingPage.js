import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import images from '../../assets/images/images';
import './LandingPage.css';
import Airplane from '../../assets/images/airplane.png';

export default class LandingPage extends Component {
  render() {
    return (
      <div className="landing-page-container-1">
        <div className="background-image">
          <img
            className="landing-page-background"
            src={images.Background}
            alt="plane soaring through sky"
          />
        </div>
        <div className="landing-page-container">
          <div className="background">
            <div className="ways-header">
              <p>Ways!</p>
            </div>
            <div className="airplane-bg">
              <img
                className="airplane-bg zoom"
                src={Airplane}
                alt="airplane taking off"
              />
            </div>
          </div>

          <section className="landing-page">
            <div className="wrapper">
              <div className="one longFade">
                <i className="fas fa-plane-departure xl fa-2x"></i>
                <hr />
                <p>Our goal is getting your vacay off the ground!</p>
              </div>
              <div className="two what-to-do longFade">
                <i className="fas fa-hotel fa-2x"></i>
                <hr />
                <p>
                  Find places to visit, or add places you know to help others!
                </p>
              </div>

              <div className="three call-to-action longFade">
                <i className="fas fa-car fa-2x"></i>
                <hr />
                <p>Plan out every stop, or take a trip another user added!</p>
              </div>
            </div>
            <div className="linkToDashButton">
              <Link to="/dashboard">
                <div className="myButton lets-go-btn">Let's Go!</div>
              </Link>
            </div>
          </section>
        </div>
      </div>
    );
  }
}
