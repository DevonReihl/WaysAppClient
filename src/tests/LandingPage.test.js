import React from 'react'
import LandingPage from '../components/LandingPage/LandingPage'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

describe(`LandingPage component`, () => {
    it('Renders without crashing', () => {
        shallow(<LandingPage />)
    })
    it('Matches snapshot', () => {
        const wrapper =  shallow(
        <LandingPage />
        )
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})