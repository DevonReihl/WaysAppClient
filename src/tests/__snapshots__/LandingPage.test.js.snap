// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage component Matches snapshot 1`] = `
<div
  className="landing-page-container-1"
>
  <div
    className="background-image"
  >
    <img
      alt="plane soaring through sky"
      className="landing-page-background"
      src="ways-landscape-2.jpg"
    />
  </div>
  <div
    className="landing-page-container"
  >
    <div
      className="background"
    >
      <div
        className="ways-header"
      >
        <p>
          Ways!
        </p>
      </div>
      <div
        className="airplane-bg"
      >
        <img
          alt="airplane taking off"
          className="airplane-bg zoom"
          src="airplane.png"
        />
      </div>
    </div>
    <section
      className="landing-page"
    >
      <div
        className="wrapper"
      >
        <div
          className="one longFade"
        >
          <i
            className="fas fa-plane-departure xl fa-2x"
          />
          <hr />
          <p>
            Our goal is getting your vacay off the ground!
          </p>
        </div>
        <div
          className="two what-to-do longFade"
        >
          <i
            className="fas fa-hotel fa-2x"
          />
          <hr />
          <p>
            Find places to visit, or add places you know to help others!
          </p>
        </div>
        <div
          className="three call-to-action longFade"
        >
          <i
            className="fas fa-car fa-2x"
          />
          <hr />
          <p>
            Plan out every stop, or take a trip another user added!
          </p>
        </div>
      </div>
      <div
        className="linkToDashButton"
      >
        <Link
          replace={false}
          to="/dashboard"
        >
          <div
            className="myButton lets-go-btn"
          >
            Let's Go!
          </div>
        </Link>
      </div>
    </section>
  </div>
</div>
`;
